### DISCUSSION TITLE
formData.get() is not a function

### DISCUSSION THREAD

**Discussion by fullstacknewb** [vercel/next.js#68175] [App Router] [answered]
Hey guys I'm having this problem with useFormState() where whenever I submit the form, for some reason the get function is not accessible. The server action works fine without useFormState(), where the get function is accessible and it retrieves the values. I am trying the display a message that was sent from the server to client.

This is my server action code
![image](https://github.com/user-attachments/assets/b208302a-8388-41b2-9f19-e63c6cd88b56)

this is my client side code 
![image](https://github.com/user-attachments/assets/dd83fce7-cd8f-4ac3-85df-4392ffda1605)

If you could help that would be great!

**Comment by Juneezee** [comment, marked as answer]
You are passing your `Register` Server Action to the `useFormState` hook. In that case the first argument will be `prevState`, and the second argument is `FormData`.

See the docs [Server-side validation and error handling](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#server-side-validation-and-error-handling) section, 

```ts
'use server'
 
export async function createUser(prevState: any, formData: FormData) {
  // ...
  return {
    message: 'Please enter a valid email',
  }
}
```

**Reply by fullstacknewb** [reply]
Thank you!

**Reply by Juneezee** [reply]
@fullstacknewb  If my answer helps you, please mark it as answer. Thanks! 

![](https://docs.github.com/assets/cb-84373/mw-1440/images/help/discussions/comment-mark-as-answer-button.webp)

**Reply by tushgaurav** [reply]
Thanks, I totally forgot about the first parameter to the server action, i was crying for 1 hour over this. Thanks! 🪅
